src/app/meetups/meetups.component.ts

```diff
@@ -64,12 +65,24 @@ export class MeetupsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.couchService.currentTime().subscribe((date) => this.dateNow = date);
   }
 
+  toggleMeetups() {
+    this.showOutdated = !this.showOutdated;
+    this.ngOnInit();

```

Comment:
We shouldn't rerun `ngOnInit` because it creates more subscriptions to the `meetupUpdated$` observable.

Instead you should add a function to the `composeFilterFunctions` with one argument which will be the meetup in the array to test.  You can make sure the filter runs by setting `this.meetups.filter = this.meetups.filter || ' '`.